File: backend-java/auth-service/src/main/java/com/cdm/auth/entity/OrganizationUserEntity.java
package com.cdm.auth.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.io.Serializable;

@Entity
@Table(name = "organization_users")
@IdClass(OrganizationUserEntity.PK.class)
@Getter @Setter @NoArgsConstructor
public class OrganizationUserEntity {

    @Id @Column(name = "org_id")
    private Long orgId;

    @Id @Column(name = "user_id")
    private Long userId;

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "user_type", length = 20)
    private String userType = "staff";

    @Getter @Setter @NoArgsConstructor
    public static class PK implements Serializable {
        private Long orgId;
        private Long userId;
    }
}

File: backend-nestjs/apps/auth/src/organization/organization-user.entity.ts
import { Entity, Column, PrimaryColumn, CreateDateColumn } from 'typeorm';

@Entity('organization_users')
export class OrganizationUserEntity {
  @PrimaryColumn({ name: 'org_id', type: 'bigint' })
  orgId: string;

  @PrimaryColumn({ name: 'user_id', type: 'bigint' })
  userId: string;

  @Column({ name: 'tenant_id', type: 'bigint' })
  tenantId: string;

  @Column({ name: 'user_type', length: 20, default: 'staff' })
  userType: string;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;
}
